File: katt/iterated/attachments/java/grader.java
import java.io.*;

public class grader {

  public static void main(String[] args) throws IOException {
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    int N = Integer.parseInt(in.readLine());
    String target = in.readLine();
    if (N != target.length()) {
      System.out.println("Invalid input string.");
      System.exit(2);
    }
    for (int i = 0; i < N; i++) {
      if (target.charAt(i) != '0' && target.charAt(i) != '1') {
        System.out.println("Invalid input string.");
        System.exit(2);
      }
    }

    lib.target = target;
    lib.N = N;
    iterated.init(N);
    System.out.println("Failure.");
    System.exit(1);
  }

  public static class lib {
    private static String target;
    private static int N, guesses;
    public static int guess(boolean[] ans) {
      assert ans.length == N;
      guesses++;
      int ret = 0;
      for (int i = 0; i < N; i++) {
        char c = target.charAt(i), d = ans[i] ? '1' : '0';
        ret += (c == d ? 1 : 0);
      }
      if (ret == N) {
        System.out.println("Success, took " + guesses + " guesses.");
        System.exit(0);
      }
      return ret;
    }
  }
}

File: katt/Contest3/iterated/attachments/python2/grader.py
import iterated
import grader
import sys

N = -1
guesses = 0
target = None
def _grader_init():
    global N
    global target
    N = int(sys.stdin.readline())
    inp = sys.stdin.readline().strip()
    if len(inp) != N:
        print "Invalid input string."
        sys.exit(2)
    if any(x != '0' and x != '1' for x in inp):
        print "Invalid input string."
        sys.exit(2)
    target = [True if x == '1' else False for x in inp]
    mastermind.init(N)
    print "Failure."
    sys.exit(1)

def guess(ans):
    assert len(ans) == N
    assert all(x == True or x == False for x in ans)
    global guesses
    guesses += 1
    res = 0
    for i in range(N):
        res += 1 if ans[i] == target[i] else 0
    if res == N:
        print "Success, took", guesses, "guesses."
        sys.exit(0)
    return res

if __name__ == "__main__":
    grader._grader_init()
